File: springboot/open-api/src/main/java/com/openapi/domain/dto/ws/base/ClientEvent.java
package com.openapi.domain.dto.ws.base;


import com.openapi.domain.constant.ws.WsChannel;
import com.openapi.domain.constant.ws.WsEvent;
import lombok.AllArgsConstructor;
import lombok.Data;
import java.util.Map;

/**
 * 客户端发送事件基类
 */
@Data
@AllArgsConstructor
public abstract class ClientEvent {
    /**
     * 事件路由，见 WsChannel 枚举
     * @see WsChannel
     */
    private String channel;
    /**
     * 事件名称，见 WsEvent 枚举
     * @see WsEvent
     */
    private String event;
    /**
     * 业务数据，key-value 格式
     */
    private Map<String, String> data;
}

File: app/android/app/src/main/java/com/magicvector/domain/dto/ws/base/ClientEvent.kt
package com.magicvector.domain.dto.ws.base


/**
 * 客户端发送事件基类
 */
open class ClientEvent(
    /**
     * 事件路由，见 WsChannel 枚举
     * @see com.magicvector.domain.constant.ws.WsEvent
     */
    open val channel: String,
    /**
     * 事件名称，见 WsEvent 枚举
     * @see com.magicvector.domain.constant.ws.WsEvent
     */
    open val event: String,
    /**
     * 业务数据，key-value 格式
     */
    open val data: Map<String, String> = emptyMap()
)